src/runtime/policy/policy.ts

```diff
@@ -279,6 +290,65 @@ export class PolicyField {
       label: label === 'raw' ? '' : label,
     };
   }
+
+  getRestrictedFields(parentType: Type) {
+    const field = parentType.getEntitySchema().fields[this.name];
+    return [this.name, this.restrictField(field)];
+  }
+
+  private restrictField(field) {
+    switch (field.kind) {
+      case 'kotlin-primitive':
+      case 'schema-primitive': {
+        assert(this.subfields.length === 0);
+        return field;
+      }
+      case 'schema-collection':
+      case 'schema-ordered-list':
+      case 'schema-nested': {
+        return FieldType.create(
+          {kind: field.kind, schema: this.restrictField(field.schema)}
+        );
+      }
+      case 'schema-reference': {
+        return FieldType.create({
+          kind: 'schema-reference', schema: {
+            ...field.schema,
+            model: this.restrictedEntityType(field.getEntityType())
+          }
+        });
+      }
+      case 'type-name':
+      case 'schema-inline': {
+        const entityType = field.getEntityType();
+        if (entityType == null) return this;
+        return FieldType.create({
+          kind: field.kind,
+          model: this.restrictedEntityType(entityType)
+        });
+      }
+      // TODO(bgogul): `field-path` does not support these types yet.
```

Comment:
Yup, this is fine, most of ProdEx and Kotlin storage stack does not support them.